docker/launch.sh

```diff
@@ -0,0 +1,374 @@
+        echo "Ubuntu version: Trusty"
+        sudo apt-get update
+        sudo apt-get -y install docker.io
+        ln -sf /usr/bin/docker.io /usr/local/bin/docker
+        sed -i '$acomplete -F _docker docker' /etc/bash_completion.d/docker.io
+        update-rc.d docker.io defaults
+    else
+        echo -e "${RED}Unsupported ubuntu version! ${NC}"
+        CLEAN_EXIT=true
+        exit 1
+    fi
+fi
+
+if [ ${NVIDIA} = true ]; then
+    sudo apt-get install -y nvidia-docker
+fi
+
+# Log in to docker only for hand h images
+function docker_login
+{
+    if [ ${HAND_H} = true ]; then
+        echo "Logging in to docker "
+        for i in 'seq 1 3';
+        do
+            if [ -z ${DOCKER_HUB_USER} ]; then
+                echo "Docker username not specified"
+                docker login
+            else
+                echo "Docker username specified"
+                if [ -z ${DOCKER_HUB_PASSWORD} ]; then
+                    docker login --username ${DOCKER_HUB_USER}
+                else
+                    docker login --username ${DOCKER_HUB_USER} --password ${DOCKER_HUB_PASSWORD}
+                fi
+            fi
+            if [ $? == 0 ]; then
+                break
+            fi
+            if [ ${i} == 3 and $? !=0 ]; then
+                echo -e "${RED}Docker login failed. You will not be able to pull private docker images.${NC}"
+                CLEAN_EXIT=true
+                exit 1
+            fi
+        done
+    fi
+}
+
+# If running for the first time create desktop shortcut
+APP_FOLDER=/home/$USER/launcher_app
+if [ ${DESKTOP_ICON} = true ] ; then
+    echo ""
+    echo " -------------------------------"
+    echo " |   Making desktop shortcut   |"
+    echo " -------------------------------"
+    echo ""
+
+    echo "Creating launcher folder"
+    if [ ! -d "${APP_FOLDER}" ]; then
+      mkdir ${APP_FOLDER}
+    fi
+
+    # Create a initial script for dexterous hand
+    if [ ${HAND_H} = false ]; then
+        if [ -z "${CONFIG_BRANCH}" ]; then
+            echo -e "${RED}Specify a config branch for your dexterous hand ${NC}"
+            CLEAN_EXIT=true
+            exit 1
+        else
+            printf "#! /bin/bash
+            source /home/user/projects/shadow_robot/base/devel/setup.bash
+            roscd sr_ethercat_hand_config
+            git fetch
+            git checkout ${CONFIG_BRANCH}
+
+            # Changing ethernet interface
+            sed -i 's|eth_port\" value=.*|eth_port\" value=\"${ETHERCAT_INTERFACE}\" />|' \$(rospack find sr_ethercat_hand_config)/launch/sr_rhand.launch
+
+            roslaunch sr_ethercat_hand_config sr_rhand.launch
+            " > ${APP_FOLDER}/setup_dexterous_hand.sh
+            chmod +x ${APP_FOLDER}/setup_dexterous_hand.sh
+        fi
+    fi
+
+    if [ -e ${APP_FOLDER}/launch.sh ]; then
+        rm ${APP_FOLDER}/launch.sh
+    fi
+    echo "Downloading the script"
+    # TODO: change this for master before merging
```

Comment:
you could create one variable with branch name on the top of the script and change is only once :)